clients/python/client/osparc/_utils.py

```diff
@@ -59,3 +59,55 @@ def __iter__(self) -> Generator[T, None, None]:
             self._offset += len(page.items)
```

Comment:
OK, this is indeed a solution. I would have instead taken advantage of the response of the pagination. The response includes already a url to the "next" page, i.e. here you are "redoing" what the pagination method does which in principle is correct :-) 
The only drawback I see is that if the pagination strategy changes, this duplication needs to be updated.

Reply:
NOTE on the asserts above.

We only use assert for development (i.e. they are removed in production). Please raise exceptions if you really want to do those checks

Reply:
Good point. I will adapt to make use of the url to the next page.